From 73e3accda680cc2edee0425ec54def5d26e8648b Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Thu, 26 Jul 2012 14:17:09 +0200
Subject: [PATCH] rhbz#842552 always create text content

The following line in oox::drawingml::table::TableCell::pushToXCell
suggests that it is assumed that the cell always contains text:

Reference< text::XText > xText( rxCell, UNO_QUERY_THROW );

So TableCell should always hold a valid TextBody too.

Change-Id: Ic2db7b535c98dd5f2b18941846709a781df4585c
---
 oox/source/drawingml/table/tablecell.cxx |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/oox/source/drawingml/table/tablecell.cxx b/oox/source/drawingml/table/tablecell.cxx
index 96fa5b7..c89a4f9 100644
--- a/oox/source/drawingml/table/tablecell.cxx
+++ b/oox/source/drawingml/table/tablecell.cxx
@@ -44,7 +44,8 @@ using ::com::sun::star::drawing::LineStyle;
 namespace oox { namespace drawingml { namespace table {
 
 TableCell::TableCell()
-: mnRowSpan ( 1 )
+: mpTextBody( new TextBody() )
+, mnRowSpan ( 1 )
 , mnGridSpan( 1 )
 , mbhMerge( sal_False )
 , mbvMerge( sal_False )
-- 
1.7.10.4

